packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -692,6 +766,93 @@ class TextMapPropagator {
     return spanContext._traceId.toString(16)
   }
 
+  _extractAwsXrayContext (carrier) {
+    if (carrier[xrayHeaderKey]) {
+      const parsedHeader = this._parseAWSTraceHeader(carrier[xrayHeaderKey])
+
+      let traceId
+      let spanId
+      let samplingPriority
+      let ddOrigin
+      const baggage = {}
+
+      if (!(
+        xrayRootKey in parsedHeader &&
+        // Regex check to ensure received header is in the same format as expected
+        // Format:
+        //   'Root=1-'
+        //   8 hexadecimal characters representing start time
+        //   '-'
+        //   24 hexadecimal characters representing trace id
+        //   ';'
+        //   'Parent='
+        //   16 hexadecimal characters representing parent span id
+        /^(?=.*Root=1-[0-9a-f]{8}-[0-9a-f]{24})(?=.*Parent=[0-9a-f]{16}).*$/i.test(carrier[xrayHeaderKey])
```

Comment:
It seems expensive that you first do a regex to see if the header matches and then later run a parser to extract values from the header. Is there a way to either use the result of the regex to capture some or all of the data, or just run the parser logic and bail if it doesn't match?

I think checking for the header like you do on line 780, removing the regex, and then doing the parsing logic starting on line 796 would be enough.